src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers.Tests/AZC0015Tests.cs

```diff
@@ -9,7 +9,7 @@ namespace Azure.ClientSdk.Analyzers.Tests
 {
     public class AZC0015Tests
     {
-        [Theory]
+        [Theory(Skip = "We use return type as an exit criteria to check method, so this test set is not applicable until we find a better criteria.")]
```

Comment:
Please don't skip tests - which of these are valid/invalid?  If a different analyzer is catching these errors now, would it make sense to update the test to detect both?

Reply:
All of these cases, we don't check them anymore. ANY method without valid return type, analyzer just skips check. So these tests are invalid until we find a better way to tell the methods we want to check. This is what you brought up, and @m-nash agrees with you.

Reply:
It looks like the result of skipping both these tests is that we don't have any tests for AZ0015, which isn't a good place for us to be for this analyzer.

Reply:
AZ0015 just not effective now. 
![image](https://github.com/Azure/azure-sdk-tools/assets/97225342/ddc39d0e-a649-4369-99c9-52b577107be2)

Reply:
Ah, I didn't realize that implementing it this way made AZC0015 completely non-functional.

I think AZC0015 is a useful diagnostic error to have running in azure-sdk-for-net.  For example, if I write a method that accidentally returns `Task<Model>` where I meant to return `Task<Response<Model>>`, I would like a warning to help me find this, and not rely on a human architect catching my mistake.

Is it possible to change the implementation for AZC0018 and AZC0019 so that AZC0015 is still functional?  I realize that a lot of services have [opted-out of this warning](https://grep.app/search?current=2&q=AZC0015&filter[repo][0]=Azure/azure-sdk-for-net), we should see if we can remove these where it's possible, but that doesn't have to be part of this PR.